org-quick-peek.el

```diff
@@ -144,6 +153,30 @@
         (unless quiet
           (minibuffer-message "Entry has no text.")))))
 
+(cl-defun org-quick-peek--agenda-show-logbook (&key quiet)
+  "Show quick peek (:LOGBOOK: drawer only) of item at current line."
+  (-if-let* ((marker (org-get-at-bol 'org-hd-marker))
+             (text (s-trim
+                    (org-quick-peek--s-trim-lines
+                    (let ((entry 
+                           (org-quick-peek--get-entry-text marker
+                                                           :num-lines nil
+                                                           :keep-drawers t)))
+                      (with-temp-buffer
```

Comment:
I think this code could be simpler.  For example, since it's necessary to parse the drawer to find its end, that could be done in the source buffer instead of in a temporary buffer.

Reply:
I used a temp buffer rather than a source buffer to avoid physically switching to the source buffer,  (and the user seeing that switch, however briefly) is there any way to simplify the existing `with-temp-buffer` approach?

Reply:
The user won't see the switch to the buffer, because `redisplay` is not called, so the display will not be updated until the command returns.  You can use `with-current-buffer` to get the drawer directly from the source buffer.

Probably the best way to handle this additional functionality is to modify `org-quick-peek--get-entry-text` to accept a function as an argument, and that function should narrow the buffer to the entry text that should be returned.  So, e.g. you could pass a partially-applied function to get a specific, named drawer, like `(apply-partially #'org-quick-peek--narrow-to-drawer "LOGBOOK")`, and that function would find the "LOGBOOK" drawer in the entry at point and narrow the buffer to it.  Then `org-quick-peek--get-entry-text` would return as much of the narrowed buffer as desired.  This would allow returning different drawers, as well as using other functions to select other content within entries.

What do you think?